fflas-ffpack/fflas/fflas_freduce.inl

```diff
@@ -544,10 +544,49 @@ namespace FFLAS { namespace details {
         }
         else
         {
-            typename Field::Element_ptr Xi = A ;
-            typename Field::ConstElement_ptr Yi = B ;
-            for (; Xi < A+m*incX; Xi+=incX, Yi += incY )
-                F.reduce (*Xi , *Yi);
+            if (m < FFLASFFPACK_COPY_REDUCE)
+            {
+                // TODO also write a ``fast'' modp that handles an incY
+                typename Field::Element_ptr Xi = A ;
+                typename Field::ConstElement_ptr Yi = B ;
+                for (; Xi < A+m*incX; Xi+=incX, Yi += incY )
+                    F.reduce (*Xi , *Yi);
+            }
+            else
+            {
+                typename Field::Element_ptr Ac;
+                typename Field::Element_ptr Bc;
```

Comment:
Why do we need Bc to be Element_ptr and not ConstElement_ptr, which would then remove the need for const_cast